### ISSUE TITLE
Gitlab Access Token by project

### ISSUE THREAD

**Issue by Blarc** [Blarc/gitlab-template-lint-plugin#105]
Currently, we support setting Gitlab's Personal Access Token for each Gitlab URL. This improves user experience, because user only needs to set the token once for each Gitlab URL (or Gitlab instance).

Should we support adding Gitlab Access Token by project? 
- This would allow better security by creating project access tokens instead of personal access tokens which can be used to access all projects. 
- It would be more cumbersome for user, because he would need to set the project access token for each new project.

**Comment by Blarc** [comment]
That's correct, but someone might not want to give an access token to a plugin, that can access sensitive or GDPR data (from the repository). We have a similar case for our dependabot, where each project needs to set up the dependabot using its own project access token.

**Comment by Weissger** [comment]
I would be such a candidate.

**Comment by Blarc** [comment]
Hey @Weissger. I am just curious what is the reason?
